### DISCUSSION TITLE
Corsair commander

### DISCUSSION THREAD

**Discussion by plasma06** [Rem0o/FanControl.Releases#597] [Ideas]
I have a corsair commander mini running all my fans that connects via USB header so none of my fans are using fan headers. (using corsair link 4 to control my fans at the moment)
The software is not detecting my fans because of this.

**Comment by Rem0o** [comment]
There is some work that has been done to get Commander Pro stuff working, but it's in its early stages and will need help from a developer who has the hardware in his hand ( I don't ).

https://github.com/LibreHardwareMonitor/LibreHardwareMonitor/pull/658

**Reply by ghost** [reply]
Hi Rem0o, I've just thrown a plugin for Command PRO together (as I hate iCUE!)

[https://github.com/iJacks1980/FanControl.CommanderPRO](https://github.com/iJacks1980/FanControl.CommanderPRO)

**Reply by Rem0o** [reply]
This looks awesome! Checked the code real quick, everything seems clean.

Out of curiosity, where did you get the protocol documentation? From liquidctl?

If everything is right, I will feature it in the README. Great job!

You could also consider contributing to https://github.com/LibreHardwareMonitor/LibreHardwareMonitor directly if you got this figured out and can test.

**Reply by ghost** [reply]
Thank you :)

Alas, I can't take credit for working out the protocol details - I found some old code a while back and was trying to make my own "Fan Control" based on Core Temp (back when NZXT's CAM couldn't get the temp of the new Ryzen CPU's).  I just found your project via JayzTwoCents, liked the look and decided to dust it off and clean it up...a lot.

I'll take a look at LibreHardwareMonitor, as I have a Commander PRO (sadly the RGB controller died so can't do anything with that) - but it's 4 in the morning right now and my bed is looking oh so comfortable ;)
